Assets/Scripts/Team/TeamController.cs

```diff
@@ -27,4 +51,52 @@ public int buy(int amount)
         
     }
 
+    [Server]
+    public void SetPlayerId(GameObject newPlayer)
+    {
+        int players = GameObject.FindGameObjectsWithTag("Player").Length - 1;
+        newPlayer.GetComponent<PlayerController>().playerId = players;
+    }
+
+    [Server]
+    public void SetPlayerTeam(GameObject newPlayer) {
+        int id = GameObject.FindGameObjectsWithTag("Player").Length - 1;
+        newPlayer.GetComponent<PlayerController>().playerId = id;
+        if(id == 0)
+        {
+            newPlayer.GetComponent<PlayerController>().teamNum = 1;
+            playersTeam1.Add(newPlayer);
+            newPlayer.GetComponent<PlayerController>().troopSpawn = GameObject.FindGameObjectWithTag("TroopSpawn1");
```

Comment:
TroopSpawn constant, are we sure this is the correct naming convention, no understanding what tower or lane this spawn responds to